lib/node/fullnode.js

```diff
@@ -631,11 +632,18 @@ class FullNode extends Node {
   /**
    * Retrieve compact filter by hash.
    * @param {Hash | Number} hash
+   * @param {Number} type
    * @returns {Promise} - Returns {@link Buffer}.
    */
 
-  async getBlockFilter(hash) {
-    if (!this.filterindex)
+  async getBlockFilter(hash, filterType) {
+    const indexers = {
+      basic: this.filterindex
+    };
+
+    const Indexer = indexers[filterType];
+
+    if (!Indexer)
```

Comment:
What if we renamed the fullnode property so that this...

https://github.com/bcoin-org/bcoin/blob/83bb3354f36db2e4576b6aa2b033f9eb242c982d/lib/node/fullnode.js#L188

...was this:

```js
      this.basicFilterindex = new FilterIndexer({
```

That might make this section a little prettier. I'm not in love with this `const indexers = ...` here, maybe we can think of a better place to iterate through all the attached indexers? Or maybe `FullNode` should have a property `indexers`